### ISSUE TITLE
ID3v2.3 genre tag parsing not working?

### ISSUE THREAD

**Issue by taylorthurlow** [krists/id3tag#18]
Here's my current situation:

	(byebug) mp3data.frames[3]
	<ID3Tag::Frames::V2::GenreFrame TCON: >
	(byebug) mp3data.frames[3].raw_content
	"\x01\xFF\xFEP\x00o\x00p\x00\x00\x00"
	(byebug) mp3data.genre
	""

This null character separated genre tag should be "Pop". I've saved this tag using mp3tag, and confirmed that it's ID3v2.3.

Am I doing something wrong here?

**Comment by krists** [comment]
Hi @taylorthurlow 

You have found a bug in the id3tag gem.

If we look at the docs https://github.com/krists/id3tag/blob/master/standard_documents/id3v2.3.0.txt#L522-L536, https://github.com/krists/id3tag/blob/master/standard_documents/id3v2.3.0.txt#L555-L573 and check  implementation https://github.com/krists/id3tag/blob/master/lib/id3tag/frames/v2/genre_frame/genre_parser_pre_24.rb#L25, https://github.com/krists/id3tag/blob/master/lib/id3tag/frames/v2/genre_frame/genre_parser.rb#L10 you can see that the genre parse tries to work with raw frame data, but it should perform text-trame related tasks like detecting encoding(first byte).

Your example has genre frame in UTF-16 encoding. When I tested this library on my music collection I did not see anything like this :)

```ruby
"\x01\xFF\xFEP\x00o\x00p\x00\x00\x00".bytes
# => [1, 255, 254, 80, 0, 111, 0, 112, 0, 0, 0]
```
first byte is 1, which means it's encoding is  UTF-16(https://github.com/krists/id3tag/blob/master/lib/id3tag/frames/v2/text_frame.rb#L8).

```ruby
"\x01\xFF\xFEP\x00o\x00p\x00\x00\x00".bytes[1..-1].pack("c*").encode(Encoding::UTF_8, Encoding::UTF_16)
# => "Pop\u0000"
```
Now this is something usable. Just need to remove termination bytes.

This functionality is already in https://github.com/krists/id3tag/blob/master/lib/id3tag/frames/v2/text_frame.rb class. So https://github.com/krists/id3tag/blob/master/lib/id3tag/frames/v2/genre_frame/genre_parser.rb could be changed to inherit from that.

@taylorthurlow Do you want to make a bugfix for this?

**Comment by krists** [comment]
@taylorthurlow Nevermind :) I have already fixed the bug and made a release `0.10.0`
